src/pdn/src/PdnGen.tcl

```diff
@@ -3614,13 +3676,69 @@ proc generate_stripes {tag net_name} {
           set template_name [dict get $plan_template $x $y]
           set layer_info [dict get $grid_data straps $lay $template_name]
           set area [list $x $y [expr $x + [dict get $template width]] [expr $y + [dict get $template height]]]
-          generate_upper_metal_mesh_stripes $tag $lay $layer_info $area
+          generate_upper_metal_mesh_stripes $tag $lay $layer_info $area $net_name
         }
       }
     }
   }
 }
 
+proc insert_power_switches {} {
+  variable grid_data
+
+  set db [ord::get_db]
+  set block [ord::get_db_block]
+
+  set psw [$db findMaster [dict get $grid_data switch_cell]]
+  # The selected power switch is double height, and has a central VSS pin, so align with VSS rails
+  set vgnd [[$psw findMTerm VGND] getMPins]
+  set vddg [[$psw findMTerm VDDG] getMPins]
+
+  set rail_lay [lindex [get_rails_layers] 0]
+  set rail_width [dict get $grid_data rails $rail_lay width]
+
+  set strap_lay [lindex [dict get $grid_data straps] 0]
+  set strap_lay_info [dict get $grid_data straps $strap_lay]
+  set strap_width [dict get $strap_lay_info width]
+
+  set area [dict get $grid_data area]
+  set offset [expr [lindex $area 0] + [dict get $strap_lay_info offset]]
+
+  set row_idx 0
+
+  foreach row [$block getRows] {
+    set orient [$row getOrient]
+    set box [$row getBBox]
+    if {$orient == "R0"} {
+      set vss_y [$box yMin]
+      set overlap_y [expr $vss_y - $rail_width / 2]
+      
+      set col_idx 0
+
+      for {set x $offset} {$x < [expr {[lindex $area 2] - [dict get $strap_lay_info width]}]} {set x [expr {[dict get $strap_lay_info pitch] + $x}]} {
+        set overlap_x [expr $x - $strap_width / 2]
```

Comment:
It might be nice to be able to control the pitch instead of relying on the strap pitch. (This could be the default if the user doesn't specify anything.

Reply:
Specifically, we're refering to the strap on the lowest metal layer, i.e. the one that connects to the stdcell rail.

The power switches connect to the vertical straps for the input (unswitched) supply, and output the switched supply on the stdcell rail. If the power switches do not align with the power straps, then there would have to be additional routing (and further IR-drop) to reach the input supply of the power switch cell and also consumes additional routing space.

Are you anticipating wanting a larger pitch between power cells, or a smaller pitch? In both cases I dont see why the pitch of the vertical power straps shouldn't also be changed in the same way, resulting once more in power switches being placced underneath the vertical strap.

FYI, The code is being adjusted to place power switches in "every other row", whether that is a "double-height" or not,  which will halve the number of power switches from the current testcase result shows.

I not convinced there is any benefit in having straps that do not connect to a power switch.

Reply:
I think in general the straps might be at a tighter pitch than the power switches, to ensure via resistances or other manufacturing issues are handled. Causing the power switches to not align with the straps perfectly.  I think it would be fine to use the default of a one to one mapping, but forcing it might make it difficult to users to actually implement something and honoring the foundries suggested strap pitches; therefore, allowing the user to override that value simply makes things more flexible.